staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -174,26 +175,33 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 	// etcd contents. Reflector framework will catch up via Watch() eventually.
 	options := metav1.ListOptions{ResourceVersion: "0"}
 	r.metrics.numberOfLists.Inc()
+	trace := utiltrace.New("Reflector_" + r.name + "_ListAndWatch")
 	start := r.clock.Now()
 	list, err := r.listerWatcher.List(options)
 	if err != nil {
 		return fmt.Errorf("%s: Failed to list %v: %v", r.name, r.expectedType, err)
 	}
+	trace.Step("Objects listed")
 	r.metrics.listDuration.Observe(time.Since(start).Seconds())
 	listMetaInterface, err := meta.ListAccessor(list)
 	if err != nil {
 		return fmt.Errorf("%s: Unable to understand list result %#v: %v", r.name, list, err)
 	}
 	resourceVersion = listMetaInterface.GetResourceVersion()
+	trace.Step("Resource version extracted")
 	items, err := meta.ExtractList(list)
 	if err != nil {
 		return fmt.Errorf("%s: Unable to understand list result %#v (%v)", r.name, list, err)
 	}
+	trace.Step("Objects extracted")
 	r.metrics.numberOfItemsInList.Observe(float64(len(items)))
 	if err := r.syncWith(items, resourceVersion); err != nil {
 		return fmt.Errorf("%s: Unable to sync list result: %v", r.name, err)
 	}
+	trace.Step("SyncWith done")
 	r.setLastSyncResourceVersion(resourceVersion)
+	trace.Step("Resource version updated")
+	trace.LogIfLong(10 * time.Second)
```

Comment:
This won't work if there will be error returned.
We should be doing defere trace.LogIfLong.

If you want to finish it here, you can pack this whole block into:
```
func() {
// here what there is now
}()
```